web/src/main/java/no/nav/foreldrepenger/fordel/web/app/rest/journalføring/FerdigstillJournalføringRestTjeneste.java

```diff
@@ -114,6 +118,153 @@
+        var journalpostId = journalpostRequest.journalpostId;
+        validerJournalpostId(journalpostId);
+
+        if (journalpostRequest.dokumenter == null || journalpostRequest.dokumenter.isEmpty() ) {
+            throw new TekniskException("FpFordel: Ingen dokumenter å oppdatere for journalpostId {}", journalpostId);
+        }
+
+        final var arkivJournalpost = hentJournalpost(journalpostId);
+        var dokumenterMedNyTittel = journalpostRequest.dokumenter;
+        final var opprinneligHovedType = arkivJournalpost.getHovedtype();
+        var dokumenterFraArkiv = arkivJournalpost.getOriginalJournalpost().dokumenter();
+
+        Set<DokumentTypeId> dokumenttyper = utledDokumentTyper(dokumenterMedNyTittel, dokumenterFraArkiv);
+
+        var nyHovedtype = ArkivUtil.utledHovedDokumentType(dokumenttyper);
+        var nyJournalpostTittel = skalOppdatereJournalpostTittel(nyHovedtype, opprinneligHovedType);
+        var dokumenterÅOppdatere = mapDokumenterTilOppdatering(dokumenterMedNyTittel);
+
+        arkivTjeneste.oppdaterJournalpostMedTitler(journalpostId, nyJournalpostTittel, dokumenterÅOppdatere);
+
+        return new JournalpostTittelDto(nyJournalpostTittel.orElse(null));
+    }
+
+    private List<no.nav.vedtak.felles.integrasjon.dokarkiv.dto.OppdaterJournalpostRequest.DokumentInfoOppdater> mapDokumenterTilOppdatering(List<OppdaterJournalpostRequest.OppdaterDokumentRequest> dokumenter) {
+        return dokumenter.stream()
+            .filter(dt -> dt.dokumentIdDto != null && dt.tittel != null)
+            .map( dt -> new no.nav.vedtak.felles.integrasjon.dokarkiv.dto.OppdaterJournalpostRequest.DokumentInfoOppdater(dt.dokumentIdDto.dokumentId(), dt.tittel(),null))
+            .toList();
+    }
+
+    public record JournalpostTittelDto(String journalpostTittel) {}
+
+    private Optional<String> skalOppdatereJournalpostTittel(DokumentTypeId nyId, DokumentTypeId gammelId) {
+        return nyId.equals(gammelId) ? Optional.empty() : Optional.of(nyId.getTermNavn());
+    }
+
+    private static Set<DokumentTypeId> utledDokumentTyper(List<OppdaterJournalpostRequest.OppdaterDokumentRequest> dokumenterMedNyTittel,
+                                                          List<DokumentInfo> dokumenterFraArkiv) {
+        Set<DokumentTypeId> dokumenttyper = new HashSet<>();
+        for(DokumentInfo dokumentFraAkriv: dokumenterFraArkiv) {
+            for (OppdaterJournalpostRequest.OppdaterDokumentRequest dokumentNyTittel : dokumenterMedNyTittel) {
+                if (dokumentFraAkriv.dokumentInfoId().equals(dokumentNyTittel.dokumentIdDto.dokumentId())) {
+                    dokumenttyper.add(DokumentTypeId.fraTermNavn(dokumentNyTittel.tittel));
+                } else {
+                    dokumenttyper.add(DokumentTypeId.fraTermNavn(dokumentFraAkriv.tittel()));
```

Comment:
Her bør vi ta en tenkerunde - siden den doble løkken vil sørge for at alt fra arkiv blir lagt til i dokumenttyper.
Mulig vi først bør lage et sett av de dokId'ene som finnes i request og arkiv.
Så bygger du opp dokumenttyper for arkiv basert på de som ikke finnes i request og de som finnes i request.
For de som kommer fra arkiv og ikke finnes i request bør man ta med brevkode i vurderingen - men ikkefor  de som kommer fra requesten
En ting å tenke på er om man skal kunne oppdatere tittel for alle dokument - også de som har tittel.